examples/run_lm_finetuning.py

```diff
@@ -223,14 +223,34 @@ def train(args, train_dataset, model, tokenizer):
     logger.info("  Total optimization steps = %d", t_total)
 
     global_step = 0
+    epochs_trained = 0
+    steps_trained_in_current_epoch = 0
+    # Check if continuing training from a checkpoint
+    if os.path.exists(args.model_name_or_path):
+        # set global_step to gobal_step of last saved checkpoint from model path
+        global_step = int(args.model_name_or_path.split('-')[-1].split('/')[0])
+        epochs_trained = global_step // (len(train_dataloader) // args.gradient_accumulation_steps)
+        steps_trained_in_current_epoch = global_step % (len(train_dataloader) // args.gradient_accumulation_steps)
+
+        logger.info("  Continuing training from checkpoint, will skip to saved global_step")
+        logger.info("  Continuing training from epoch %d", epochs_trained)
+        logger.info("  Continuing training from global step %d", global_step)
+        logger.info("  Will skip the first %d steps in the first epoch", steps_trained_in_current_epoch)
+
     tr_loss, logging_loss = 0.0, 0.0
     model.resize_token_embeddings(len(tokenizer))
     model.zero_grad()
-    train_iterator = trange(int(args.num_train_epochs), desc="Epoch", disable=args.local_rank not in [-1, 0])
+    train_iterator = trange(epochs_trained, int(args.num_train_epochs), desc="Epoch", disable=args.local_rank not in [-1, 0])
```

Comment:
Hi, I have a problem here.
Indeed, to keep the way of sampling data will be the same as before, we need to set the random seed as the same and iterate over the trained samples. So I think we should iterate the `train_data_loader` for `epochs_trained` times, not just skip them. Because the dataloader samples data differently in different epoch.

Reply:
I ran some experiments [here](https://app.wandb.ai/bkkaggle/saving-and-resuming?workspace=default), it looks like there is some amount of stochasticity even when interrupting and resuming training within a single epoch, so I'm not sure if changing this and just continuing through all the epochs & steps will solve the problem

Reply:
Thank you very much. I may test it later.